src/utils/matchers/rule_matchers.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { DirectoryMatchingRule, UserInfo } from "../types";
+/* eslint-enable @typescript-eslint/no-unused-vars */
+import { MATCH_OPTIONS } from "../config";
+import Mustache from "mustache";
+import minimatch from "minimatch";
+
+/**
+ * Matches a single file against a single rule to check for ownership.
+ *
+ * @param rule The user defined rule to identify file ownership.
+ * @param filename The file in the pull request that should be checked.
+ * @param info The user information that can be used in rules.
+ * @param context The request context from Probot core.
+ */
+export const matchRule = (
+  rule: DirectoryMatchingRule,
+  filename: string,
+  info: UserInfo,
+  log_msg: (msg: string) => void,
```

Comment:
Codacy found an issue: [Identifier 'log_msg' is not in camel case.](https://app.codacy.com/manual/tianhaoz/approveman/pullRequest?prid=6318502)